File: src/Explosion.java
import javax.swing.*;
import java.awt.*;

/** Explosion class for jWars */
public class Explosion extends Entity {
	
	private Entity source;
	private Color currentColor;

	/** Standard constructor creates a new explosion from the given source
	 *  @param src source of the explosion */
	public Explosion(Entity src) {
		super();	
		setSource(src);
		setEntityWidth(1);
		setXpos(src.getXpos());
		setYpos(src.getYpos());	
		currentColor = Color.ORANGE;
	}
	
	/** Paint the explosion
	 *  @param g graphics object to paint on */
	public void paintComponent(Graphics g) {
		g.setColor(currentColor);
		if(getEntityWidth() < source.getEntityWidth()) {
			g.fillOval(getXpos(),getYpos(),getEntityWidth(),getEntityWidth());
			setEntityWidth(getEntityWidth() + 2);
			if(currentColor.equals(Color.ORANGE)) {
				currentColor = Color.RED;
			}
			else {
				currentColor = Color.ORANGE;
			}
		}	
	}

	/** Set the source of the explosion
	 *  @param src source of the explosion */	
	public final void setSource(Entity src) {
		source = src;
	}

	/** Get the source of the explosion
	 *  @return explosion source */
	public final Entity getSource() {
		return source;
	}
}

File: src/Explosion.scala
import javax.swing._
import java.awt._

/** Explosion class for jWars */
class Explosion(src: Entity) extends Entity {
	
	var source = src
	var currentColor = Color.ORANGE

	entityWidth = 1
	xPos = src.xPos
	yPos = src.yPos
	
	/** Paint the explosion
	 *  @param g graphics object to paint on */
	override def paintComponent(g: Graphics) = {
		g.setColor(currentColor);
		if(entityWidth < source.entityWidth) {
			g.fillOval(xPos,yPos,entityWidth,entityHeight)
			entityWidth = entityWidth + 2
			entityHeight = entityHeight + 2
			if(currentColor.equals(Color.ORANGE)) {
				currentColor = Color.RED
			}
			else {
				currentColor = Color.ORANGE
			}
		}	
	}

}
